### DISCUSSION TITLE
Restarting active development

### DISCUSSION THREAD

**Discussion by dstpierre** [staticbackendhq/core#116] [Announcements]
Hey

It's been almost a full year since I've released the [latest version](https://github.com/staticbackendhq/core/releases/tag/v1.5.0) of StaticBackend.

I won't lie, after failing at trying to find a commercial path for SB and alternatively trying to find funding I kind of stop active development.

Getting traction is also difficult, but I believe I'm to blame for that. The project isn't the easiest to get started with and there's certainly a lack of examples. Keeping an open source project alive when there's not much reactions is hard.

That being said, I still firmly believe in what we've build here and I think it can reach the next level in the following year or so. I'm missing working on SB, which at this point is already a decently sized Go project in my opinion.

### Where are things now?

As someone that used StaticBackend in production for two SaaS, I think its API is workable. I think the problem it tries to solve is still very present which is, handling most of the common backend tasks of a typical web applications from a single executable file with optional external database servers.

It might not be the easiest to get started with, maybe it's messaging is a bit off and people does not really know what they'll get into or how it might fit with their use cases.

### Where it's heading

Here's what I'd like to achieve in the following year or so:

1. Keep the public API aka no v2 is planned.
2. Extract code from `core` by creating plugins for things that aren't that common, like html/image to PDF, etc.
3. Make the `core` smaller and trim its dependencies. At this time the CLI for local dev is around 170 MB.
4. Change messaging where SB is an event-driven self-hosted backend server that helps people build quick v1 of their app.
5. Create lots of tutorial example in text and video format, maybe a free course on creating an app with SB.
6. Improving and cleaning existing code. Improving tests.
7. Try to get more traction with this open source project.
8. Make the project evolves based on community feedback.
9. Streamline the release process
10. Have more option to subscribe to certain data on realtime events.

In any case, this time there's no commercial agenda, I'm just happy to work back on my project and still very proud of its current state.

I'm keeping the Discord. I'll try to keep all channels more active, starting from now.

Thanks and do not hesitate to participate, I'm always happy to talk.
Dominic

**Comment by g41797** [comment]
What do you think regarding

- go based _Server-side functions_ 
- Extra functions

both above actually go plugins

**Comment by g41797** [comment]
serverless SB functions/services - place custom code out of the SB process

i'd like to develop this part
just for fun

opened #128

**Comment by dstpierre** [comment]
Hey @g41797 crazy late response, I kind of archived the project in 2024 without replying to pending items, I thought I'd still can reply after archiving on existing issues/PRs/duscussions, turns out nope.

Having Go based server-side function would be great, I even thought about having them being WASM with a wasm runner, that way all languages capable of building to WASM could be used as server-side language.

I'm restarting the project btw, sorry for not replying to you back then.

**Comment by dstpierre** [comment]
I'll close this and restart a new announcement as instead of going forward in 2024 I archived the project.
